src/rich-text/plugins/placeholder.ts

```diff
@@ -0,0 +1,40 @@
+import { Fragment } from "prosemirror-model";
+import { Plugin } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+
+function showPlaceholder(content: Fragment<any>, textContent: string) {
+    const { firstChild } = content;
+    const { name } = firstChild.type;
+    const allowPlaceholder =
+        name === "paragraph" ||
+        name === "heading" ||
+        firstChild.attrs.params === "markdown";
+    // TODO check for image upload placeholder
+    return !textContent && allowPlaceholder && firstChild.childCount === 0;
+}
+
+// TODO write a test for this
+/** Plugin that add placeholder text to the editor when it's empty */
+export function placeholderPlugin(placeholder = ""): Plugin {
```

Comment:
I'd not make this param optional:

```suggestion
export function placeholderPlugin(placeholder: string): Plugin {
```